### PR TITLE
=doc #23421 move Kill docs around and fix snippet

### LINKED ISSUE #23421: Scala Kill Documentation
In the Actors documentation for [Killing an Actor](http://doc.akka.io/docs/akka/2.5.3/scala/actors.html#killing-an-actor), the code example for Scala is given as:

```
victim ! "kill"
expectMsg("finished")
```
This is just a plain text message that's handled explicitly by receive, and doesn't convey the use of the Kill instance that is the purpose of that section: http://doc.akka.io/api/akka/current/akka/actor/Kill$.html

I would update this myself, but the example snippet seems to be part of the Akka tests, and I'm not familiar enough with the test suite to make the changes.

### PR DESCRIPTION
**Pull request by ktoso**
resolves #23421

### CODE REVIEW

File: akka-docs/src/test/scala/docs/actor/ActorDocSpec.scala [hunk 1]
```diff
@@ -586,15 +587,28 @@ class ActorDocSpec extends AkkaSpec("""
...
+
+    //#kill
+    context.watch(victim) // watch the Actor to receive Terminated message once it dies
+    
+    victim ! Kill
+
+    expectMsg(Terminated(victim)) // the Actor has indeed terminated
+    //#kill
+  }
+
```
Comment (johanandren) [inline akka-docs/src/test/scala/docs/actor/ActorDocSpec.scala]: What about the Java Sample?

File: akka-docs/src/test/scala/docs/actor/ActorDocSpec.scala [hunk 2]
```diff
@@ -586,15 +587,28 @@ class ActorDocSpec extends AkkaSpec("""
...
+    val victim = system.actorOf(TestActors.echoActorProps)
+    implicit val sender = testActor
+    val context = this
+
+    //#kill
+    context.watch(victim) // watch the Actor to receive Terminated message once it dies
+    
+    victim ! Kill
+
+    expectMsg(Terminated(victim)) // the Actor has indeed terminated
```
Comment (raboof) [inline akka-docs/src/test/scala/docs/actor/ActorDocSpec.scala]: I guess we need `expectMsgPF` here

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
something small to complete this PR?

**Comment by ktoso** [comment]
Finishing this now

File: akka-docs/src/test/scala/docs/actor/ActorDocSpec.scala (hunk 1 above)
Comment (ktoso) [reply akka-docs/src/test/scala/docs/actor/ActorDocSpec.scala]: it existed and was OK :)

**Pull request merged**